### PR TITLE
neovim => 0.7.2 + relevant updates

### LINKED ISSUE #7367: nvim: error while loading shared libraries: libiconv.so.2: cannot open shared object file: No such file or directory
## Description

Error when trying to open neovim.

```shell
chronos@localhost / $ nvim
```

Output:
```
nvim: error while loading shared libraries: libiconv.so.2: cannot open shared object file: No such file or directory
```

## Versions and system information
***
Ignoring debug-1.4.0 because its extensions are not built. Try: gem pristine debug --version 1.4.0
- Architecture: `x86_64` (`x86_64`)
- Kernel version: `4.19.241-13212-ge9b0bab46475`

- Chromebrew version: `1.25.3`
- Chromebrew prefix: `/usr/local`
- Chromebrew libdir: `/usr/local/lib64`

- Last update in local repository: d0cbc01 `Update self.preflight in openjdk8 (#7365) (24 hours ago)`

- OS variant: `Chrome OS`
- OS version: `cyan-release/R103-14816.131.0`
- OS channel: `stable-channel`
***

## Additional information
I don't think its related but any crew command I do will spit out the following then do its command
```shell
Ignoring debug-1.4.0 because its extensions are not built. Try: gem pristine debug --version 1.4.0
```

### PR DESCRIPTION
**Pull request by satmandu**
Fixes #7367

- update neovim to current
- update luajit to a repo which hasn't been abandoned
- fix lua library paths and update package
- add necessary dependencies for neovim, with relevant Lua packages built against luajit.

Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l`

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=neovim  CREW_TESTING=1 crew update
```
image

### CODE REVIEW

File: packages/luajit.rb [hunk 1]
```diff
@@ -2,33 +2,32 @@
...
-     armv7l: '0518de335a3dc8ac04d9bcbe476d3cf2f053c058737d57f6d723c31c6eb59c99',
-       i686: '4942a96c905bdbc35026f9737cf55e7ca9879f06bfe9923229aeeb51039fe96b',
-     x86_64: 'd95af6ff12ca102dfad963a0606b5e2181f9e3fd68788b470f21de580f52142b'
+    aarch64: '18fbd6916f9f75db3b5939997af0ef447102f49439941852c044e079d82bb611',
+     armv7l: '18fbd6916f9f75db3b5939997af0ef447102f49439941852c044e079d82bb611',
+       i686: '747299199b1011f83b77c0e142052811bb0f4736a1ec834062ff6f1c36338951',
+     x86_64: 'ad76e59f1079539a74159d81563ed7d23f5e333661d63a46cc3f6508d8fd8c5e'
   })
 
+  git_hashtag "v#{version}"
```
Comment (uberhacker) [inline packages/luajit.rb:24]: Can we move this line above the binaries below source_url?

File: packages/luajit_bitop.rb [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+# https://github.com/archlinux/svntogit-community/raw/packages/lua-bitop/trunk/PKGBUILD
+
+require 'package'
+
+class Luajit_bitop < Package
+  description 'Bitwise operations on numbers'
+  homepage 'https://bitop.luajit.org/'
+  version '1.0.2'
+  license 'MIT'
+  compatibility 'all'
```
Comment (uberhacker) [inline packages/luajit_bitop.rb:11]: `compatibility 'x86_64'`?

**Review by uberhacker — CHANGES_REQUESTED** [review]
Add new packages to packages.yaml?

File: packages/luajit_bitop.rb (hunk 2 above)
Comment (satmandu) [reply packages/luajit_bitop.rb:11]: Oops. That's a build issue... My

File: tools/packages.yaml [hunk 3]
```diff
@@ -3879,6 +3879,11 @@ url: https://www.gnupg.org/ftp/gcrypt/libksba
 activity: none
 ---
 kind: url
+name: libluv
+url: https://github.com/luvit/luv
```
Comment (uberhacker) [inline tools/packages.yaml:3883]: The url here is supposed to indicate where the latest release can be downloaded.  For github projects, that is normally releases or tags.  The original intention of this file was to provide a manifest for package updates.

File: packages/luajit_mpack.rb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  })
+  binary_sha256({
+    aarch64: '1be27ef45bfaf01f91a0be5ff9a4515224176a0c480bc1b475b4ab9e2519f46b',
+     armv7l: '1be27ef45bfaf01f91a0be5ff9a4515224176a0c480bc1b475b4ab9e2519f46b',
+       i686: '86fb45f1d5eb2ed5edb9bd291bf05f4d58730007e6adb067d544a26feb01bf3e',
+     x86_64: '4f23373e477afd0e99dc27cce8bbb8340478f98b1a96b30cdc43740523d11422'
+  })
+
+  depends_on 'libmpack'
+  depends_on 'luajit' => ':build'
```
Comment (uberhacker) [inline packages/luajit_mpack.rb:29]: Change `':build'` to `:build`?

**Review by uberhacker — CHANGES_REQUESTED** [review]
I probably should have explained the purpose of packages.yaml earlier.

File: tools/packages.yaml (hunk 3 above)
Comment (satmandu) [reply tools/packages.yaml:3883]: So what should this be?

File: packages/luajit_mpack.rb (hunk 4 above)
Comment (satmandu) [reply packages/luajit_mpack.rb]: Done

File: tools/packages.yaml (hunk 3 above)
Comment (uberhacker) [reply tools/packages.yaml:3883]: `https://github.com/luvit/luv/releases`
Note: This is not the case for all github projects.  Sometimes there is only `/tags` or nothing at all.

Comment (satmandu) [reply tools/packages.yaml]: I made all the changes I could here...

**Review by uberhacker — APPROVED** [review]

**Pull request merged**